SmartDeviceLink/SDLIAPTransport.m

```diff
@@ -249,137 +282,133 @@ - (void)connect {
-        return MultiSessionProtocolString;
-    }
 
-    if (![protocolStrings containsObject:LegacyProtocolString]) {
-        return LegacyProtocolString;
-    }
-
-    for (int i = 0; i < 30; i++) {
-        NSString *indexedProtocolString = [NSString stringWithFormat:@"%@%i", IndexedProtocolStringPrefix, i];
-        if (![protocolStrings containsObject:indexedProtocolString]) {
-            return indexedProtocolString;
-        }
-    }
-
-    return nil;
+    return connecting;
 }
 
 /**
- *  Attempt to establish a session with an accessory, or if nil is passed, to scan for one.
+ *  Attempts to establish a session with the passed accessory. If nil is passed the accessory manager is checked for connected SDL enabled accessories.
  *
  *  @param accessory The accessory to try to establish a session with, or nil to scan all connected accessories.
  */
 - (void)sdl_establishSessionWithAccessory:(nullable EAAccessory *)accessory {
     SDLLogD(@"Attempting to connect accessory: %@", accessory.name);
     if (self.retryCounter < CreateSessionRetries) {
         self.retryCounter++;
-        
+
         EAAccessory *sdlAccessory = accessory;
-        // If we are being called from sdl_connectAccessory, the EAAccessoryDidConnectNotification will contain the SDL accessory to connect to and we can connect without searching the accessory manager's connected accessory list. Otherwise, we fall through to a search.
+        // If called from sdl_connectAccessory, the notification will contain the SDL accessory to connect to and we can connect without searching the accessory manager's connected accessory list. Otherwise, we fall through to a search.
         if (sdlAccessory != nil && [self sdl_connectAccessory:sdlAccessory]) {
             // Connection underway, exit
             SDLLogV(@"Connection already underway");
             return;
         }
 
-        if ([self.class sdl_supportsRequiredProtocolStrings] != nil) {
-            NSString *failedString = [self.class sdl_supportsRequiredProtocolStrings];
-            SDLLogE(@"A required External Accessory protocol string is missing from the info.plist: %@", failedString);
-            NSAssert(NO, @"Some SDL protocol strings are not supported, check the README for all strings that must be included in your info.plist file. Missing string: %@", failedString);
+        if (![self.class sdl_plistContainsAllSupportedProtocolStrings]) {
             return;
         }
-        
+
         // Determine if we can start a multi-app session or a legacy (single-app) session
         if ((sdlAccessory = [EAAccessoryManager findAccessoryForProtocol:MultiSessionProtocolString]) && SDL_SYSTEM_VERSION_GREATER_THAN_OR_EQUAL_TO(@"9")) {
-            [self sdl_createIAPDataSessionWithAccessory:sdlAccessory forProtocol:MultiSessionProtocolString];
+            self.dataSession = [self sdl_createDataSessionWithAccessory:sdlAccessory forProtocol:MultiSessionProtocolString];
```

Comment:
Instead of this, should it maybe just call `sdl_connectAccessory`? It's kind of duplicating code.

Reply:
Refactored.